Gimbal plugin

addy1997 opened the pull request:
## Summary

A model plugin to control the `gimbal` added. This plugin has been adapted from the `gimbal controller` from this [repo](https://github.com/utiasSTARS/PhoenixDrone-sitl_gazebo/blob/master/include/gazebo_gimbal_controller_plugin.hh) and the `gimbal_small_2d` plugin from the `Gazebo (classic)`.

## Test it

* SDF file and meshes for the plugin can be found [here](https://github.com/addy1997/ign-gazebo/tree/gimbal-plugin/examples/plugin/gimbal_controller/gimbal). 

* In order to test the plugin, please use the gimbal model found [here](https://app.ignitionrobotics.org/OpenRobotics/fuel/models/Gimbal%20Small%202D).

* Broadly, the idea is to attach a gimbal to the base link of a drone and control it. I am still working on it and will add the files for the model on the fuel shortly.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [x] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

ahcorde reviewed (changes requested):
Hi @addy1997, 

 - Can you explain why you are including the 3D files and the sdf inside the plugin folder. The right way to do it, it's to create a model in [ignition fuel](https://app.ignitionrobotics.org/dashboard), create an example world and fetch the model.
 - In general you should review the style, the indentation looks bad.
 - The year in the Licence should be 2021

Can you add some documentation about how to use it ?

addy1997 commented:
As the gimbal model already exists on fuel, I will remove the sdf file from
my commit, fix the indentation and add documentation.

On Mon, 31 May 2021, 13:26 Alejandro Hernández Cordero, <
***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> Hi @addy1997 <https://github.com/addy1997>,
>
>    - Can you explain why you are including the 3D files and the sdf
>    inside the plugin folder. The right way to do it, it's to create a model in ignition
>    fuel <https://app.ignitionrobotics.org/dashboard>, create an example
>    world and fetch the model.
>    - In general you should review the style, the indentation looks bad.
>    - The year in the Licence should be 2021
>
> Can you add some documentation about how to use it ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ignitionrobotics/ign-gazebo/pull/840#pullrequestreview-672052112>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHALNOT54V4X554PTJYL6ZLTQM6K5ANCNFSM45WMY6XQ>
> .
>

addy1997 commented:
> Hi @addy1997,
> 
>     * Can you explain why you are including the 3D files and the sdf inside the plugin folder. The right way to do it, it's to create a model in [ignition fuel](https://app.ignitionrobotics.org/dashboard), create an example world and fetch the model.
> 
>     * In general you should review the style, the indentation looks bad.
> 
>     * The year in the Licence should be 2021
> 
> 
> Can you add some documentation about how to use it ?
@ahcorde,
1. I have fixed the indentation and reduced extra blank lines.
2. I have removed the mesh files and the model files.

ahcorde commented on examples/plugin/gimbal_controller/GimbalControllerPlugin.cc:
```diff
@@ -0,0 +1,487 @@
+      }
+    }
+}
+////////////////////////////////////////////////////////////////////////////
+
```
```suggestion

////////////////////////////////////////////////////////////////////////////
```

ahcorde commented on examples/plugin/gimbal_controller/GimbalControllerPlugin.hh:
```diff
@@ -0,0 +1,67 @@
+
+    /// \brief A plugin that simulates lift and drag.
+    class GimbalControllerPlugin
+
```
```suggestion
```

ahcorde commented on examples/plugin/gimbal_controller/GimbalControllerPlugin.hh:
```diff
@@ -0,0 +1,67 @@
+
+        /// \brief Private data pointer
+        private: std::unique_ptr<GimbalControllerPluginPrivate> dataPtr;
+
```
```suggestion
```

ahcorde reviewed (changes requested):
Indentation is still wrong, In same places you are using tabs, in others 2 spaces, 4 spaces or other values.

Please use always spaces and use the same number of spaces for each indentation level, for example 2 spaces.

addy1997 commented:
> Indentation is still wrong, In same places you are using tabs, in others 2 spaces, 4 spaces or other values.
> 
> Please use always spaces and use the same number of spaces for each indentation level, for example 2 spaces.

@ahcorde, I hope this follows the guidelines.

ahcorde reviewed (changes requested):
you should clean the history of the PR, you are including some commits that are not part of your contribution

addy1997 commented:
@ahcorde, I have pretty much badly messed up this branch. I will start from a fresh branch at some other time. Thanks and apologies for the inconvenience.
